### PR TITLE
PR for Issue #5 | Added constants file for hard-cored variables

### PR DESCRIPTION
**Pull request by kailash360**
I have created 2 constants file - one for client and the other for server where I have added the hard-cored variables.
fixes issue #5

### CODE REVIEW

File: client/src/constants.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module.exports = {
+  PREFIX: 'na-rce-',
+  SOCKET: "http://localhost:9000",
```
Comment (nafees87n) [inline client/src/constants.js:3]:
```suggestion
  SERVER_URL: "http://localhost:9000",
```
Let's call it `SERVER_URL`

File: server/constants.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    BASIC_REGEX: /import\s+(.+)/gim,
+    SPECIFIC_REGEX: /import.+from\s+(.+)/gim,
+    REJECT_LIST: ['os', 'subprocess', 'shlex', 'xml', 'pickle'],
+  },
+
+  /*CONSTANTS FOR SOCKET*/
+  SOCKET: {
+    ORIGIN: 'http://localhost:3000',
+    METHODS: ['GET', 'POST'],
+    HEADERS: ['my-custom-header']
```
Comment (nafees87n) [inline server/constants.js:50]: Remove `HEADERS`, we are not using it

File: server/socket/index.js [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
 
 module.exports.initializer = (socketIo, server) => {
   const io = socketIo(server, {
     cors: {
-      origin: 'http://localhost:3000',
-      methods: ['GET', 'POST'],
-      allowedHeaders: ['my-custom-header'],
+      origin: SOCKET.ORIGIN,
+      methods: SOCKET.METHODS,
+      allowedHeaders: SOCKET.HEADERS,
```
Comment (nafees87n) [inline server/socket/index.js:9]: remove `allowedHeaders`

File: server/constants.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+module.exports = {
+  PORT: 9000,
+  USERSTORAGEPATH: '/storage/',
+
```
Comment (nafees87n) [inline server/constants.js:4]: Add one more key `CLIENT_URL:'http://localhost:3000'`

File: client/src/components/Homepage/index.js [hunk 5]
```diff
@@ -18,16 +18,14 @@ import ModalBox from '../Modalbox/ModalBox'
 import '../../fonts/JetBrainsMono[wght].ttf'
 import './Homepage.css'
 
+//constants
+const {SOCKET,MODES,DEFAULT_CODE,DOCS,GITHUB_URL} = require('../../constants.js')
```
Comment (nafees87n) [inline client/src/components/Homepage/index.js:22]: Do not destructure the CONSTANTS while import. Suppose in future, we need to add more constants then the variable has to be imported separately as its restructured.
So can you please change all the constant imports to `const CONSTANTS=require('../../constants.js')`

File: client/src/components/Homepage/index.js [hunk 6]
```diff
@@ -18,16 +18,14 @@ import ModalBox from '../Modalbox/ModalBox'
...
-const socket = openSocket('http://localhost:9000')
-const modes = { javascript: 'js', c_cpp: 'cpp', python: 'py' }
-const defaultCode = {
-  javascript: "console.log('hello rce')",
-  c_cpp:
-    '#include <iostream>\n\nint main() {\n\tstd::cout << "hello rce";\n\treturn 0;\n}',
-  python: "print('hello rce')",
-}
+var languages = ["python", "c_cpp", "javascript"];
+const socket = openSocket(SOCKET);
```
Comment (nafees87n) [inline client/src/components/Homepage/index.js:26]: Here we could then use `CONSTANTS.SOCKET`

**Review by nafees87n — CHANGES_REQUESTED** [review]
Added some suggestions in the code. After that, it looks good to merge.
Please change the constant imports for all the instances and do remember to test the app after that.

**Comment by kailash360** [comment]
Sir, I have made the changed as requested.

**Review by nafees87n — APPROVED** [review]

**Comment by nafees87n** [comment]
Good work @kailash360 
Thanks for the contribution. Merged into `main`

**Pull request merged**